constrain/library/G36FreezeProtectionStage3.py

```diff
@@ -1,80 +1,122 @@
+
+### Code requirement
+
+- Code Name: ASHRAE Guideline 36
+- Code Year: 2021
+- Code Section: 5.16.12 Freeze Protection
+- Code Subsection: 5.16.12.3 Stage 3 (Highest)
+
+### Verification Approach
+
+The verification monitors multiple conditions that can trigger stage 3 protection: freeze-stat signal, sustained low temperatures, or critically low temperatures. When triggered, it verifies that fans are stopped, dampers are closed, and coils are properly positioned to prevent freezing damage.
 
-### Verification logic
+### Verification Applicability
+
+- Building Type(s): any
+- Space Type(s): any
+- System(s): Air handling units
+- Climate Zone(s): any
+- Component(s): supply air temperature sensors, freeze-stats, fans, dampers, coils
+
+### Verification Algorithm Pseudo Code
 
 ```python
 if supply_air_temp < 3.3 (continuously 15 minutes) or
   supply_air_temp < 1 (continuously 5 minutes) or
-  freeze_stat == True
-  if not (
-    outdoor_damper_command == 0 and
-    supply_fan_status == 'off' and
-    return_fan_status == 'off' and
-    relief_fan_status == 'off' and
-    cooling_coil_command == 100 and
-    heating_coil_command > 0
-  ):
-    fail
-  else:
-    pass
+  freeze_stat == True:
+    if not (
+        outdoor_damper_command == 0 and
+        supply_fan_status == 'off' and
+        return_fan_status == 'off' and
+        relief_fan_status == 'off' and
+        cooling_coil_command == 100 and
+        heating_coil_command > 0
+    ):
+        fail
+    else:
+        pass
 
 if never (
-  supply_air_temp < 3.3 (continuously 15 minutes) or
-  supply_air_temp < 1 (continuously 5 minutes) or
-  freeze_stat == True
+    supply_air_temp < 3.3 (continuously 15 minutes) or
+    supply_air_temp < 1 (continuously 5 minutes) or
+    freeze_stat == True
 ):
-  untested
-
+    untested
 ```
 
 ### Data requirements
 
-- freeze_stat: (optional, set to False if system does not have it) binary freeze-stat
-- supply_air_temp: supply air temperature
-- outdoor_damper_command: outdoor air damper
-- supply_fan_status: supply fan status (speed): [1, 0] (can be replaced by binary or numeric variables)
-- return_fan_status: (optional, set to False if system does not have it) return fan status (speed)
-- relief_fan_status: (optional, set to False if system does not have it) relief fan status (speed)
-- cooling_coil_command: cooling coil command
-- heating_coil_command: heating coil command
+- freeze_stat: Freeze protection flag
```

Comment:
wrong description

Reply:
I brought the description from the library file ((optional, set to False if system does not have it) Binary freeze-stat)